plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,460 @@
+import org.gradle.api.tasks.Optional
+import org.gradle.execution.RunRootBuildWorkBuildOperationType
+import org.gradle.execution.taskgraph.NotifyTaskGraphWhenReadyBuildOperationType
+import org.gradle.internal.operations.BuildOperationDescriptor
+import org.gradle.internal.operations.BuildOperationListener
+import org.gradle.internal.operations.OperationFinishEvent
+import org.gradle.internal.operations.OperationIdentifier
+import org.gradle.internal.operations.OperationProgressEvent
+import org.gradle.internal.operations.OperationStartEvent
+import org.gradle.process.ExecOutput
+import org.gradle.util.GradleVersion
+
+abstract class SentryTelemetryService :
+    BuildService<SentryTelemetryService.Params>,
+    BuildOperationListener,
+    AutoCloseable {
+    interface Params : BuildServiceParameters {
+        @get:Input
+        val sendTelemetry: Property<Boolean>
+
+        @get:Input
+        val dsn: Property<String>
+
+        @get:Input
+        @get:Optional
+        val sentryOrganization: Property<String>
+
+        @get:Input
+        @get:Optional
+        val defaultSentryOrganization: Property<String>
+
+        @get:Input
+        val buildType: Property<String>
+
+        @get:Input
+        val extraTags: MapProperty<String, String>
+
+        @get:Input
+        val saas: Property<Boolean>
+
+        @get:Input
+        val cliVersion: Property<String>
+    }
+
+    private val hub: IHub
+    private var transaction: ITransaction? = null
+    private var configPhaseSpan: ISpan? = null
+    private var executionPhaseSpan: ISpan? = null
+    private var didAddChildSpans: Boolean = false
+
+    init {
+        // TODO duplicate transaction SentryPlugin and SentryJvmPlugin
+        // TODO config phase errors are not reported because this Service is never initialized
+        if (!parameters.saas.get()) {
+            println(
+                "Sentry is running against a self hosted instance. Telemetry has been disabled."
+            )
+            hub = NoOpHub.getInstance()
+        } else if (!parameters.sendTelemetry.get()) {
+            println("Sentry telemetry has been disabled.")
+            hub = NoOpHub.getInstance()
+        } else {
+            if (!Sentry.isEnabled()) {
+                // TODO test telemetry disabled
+                // TODO Logs a message, telemetry is enabled and how to disable it
+                Sentry.init { options ->
+                    options.dsn = parameters.dsn.get()
+                    options.isDebug = true
+                    options.isEnablePrettySerializationOutput = false
+                    options.tracesSampleRate = 1.0
+                    options.release = BuildConfig.Version
+                    options.isSendModules = false
+                    options.environment = parameters.buildType.get()
+                    options.setTag("SDK_VERSION", BuildConfig.SdkVersion)
+                    options.setTag("AGP_VERSION", BuildConfig.AgpVersion)
```

Comment:
`l`: we should actually read AGP version from the runtime classpath ehre, because `BuildConfig.AgpVersion` is what we package within the plugin, but since we use `compileOnly` for that, it will be overridden by whichever version users have on their classpath. You can get it via `AgpVersions.CURRENT`